version lock go vet

stevekuznetsov opened the pull request:
@liggitt @bparees @derekwaynecarr 

`go get` will always fetch the latest `HEAD`, so it seems in order to version-lock the tool we should just `git checkout` as was pointed out to me on #go-nuts. Thoughts?

bparees commented:
seems reasonable enough to me.

liggitt commented on lib/vagrant-openshift/action/install_openshift_base_dependencies.rb:
```diff
@@ -109,6 +109,7 @@ def call(env)
 fi
 
 GOPATH=/data go get golang.org/x/tools/cmd/vet
+cd /data/src/golang.org/x/tools && git checkout c262de870b618eed648983aa994b03bc04641c72 && cd -
```
pushd/popd might be more consistent with other places

liggitt replied:
Probably a comment as to why this is needed would be good as well. Is this particular version a "known good" version?

stevekuznetsov commented:
@liggitt addressed your comments. This is not a particularly special commit, but after https://github.com/openshift/origin/pull/4683 merges, it will be a version that is completely supported in Origin and thereby becomes special.

liggitt commented:
That's fine. I didn't know if particular commits to HEAD for go vet were responsible for letting in bad Printf calls, etc.

liggitt commented:
LGTM

liggitt commented:
@danmcp can you merge this?

liggitt commented:
actually, hang on... @stevekuznetsov do you need to rebuild `go vet` after checking out the version we want to lock? `go get` clones and builds, and I think you're missing the build step

liggitt commented on lib/vagrant-openshift/action/install_openshift_base_dependencies.rb:
```diff
@@ -110,6 +110,11 @@ def call(env)
 
+# Check out a stable commit for go vet in order to version lock it to something we can work with
+pushd /data/src/golang.org/x/tools >/dev/null 2>&1
+git checkout c262de870b618eed648983aa994b03bc04641c72 
```
indent and add a build call

stevekuznetsov commented:
@liggitt Good catch, we'll need to `go build` and `go install` I believe.

stevekuznetsov commented:
@liggitt Also notable that this is going to affect all tools, not just `go vet`.

liggitt commented on lib/vagrant-openshift/action/install_openshift_base_dependencies.rb:
```diff
@@ -110,6 +110,14 @@ def call(env)
+git checkout c262de870b618eed648983aa994b03bc04641c72 
+popd >/dev/null 2>&1
+# Re-install using this version of the tool
+go install golang.org/x/tools/cmd/vet
```
don't you need to set GOPATH=/data here as well? wonder if you should just export it above for all commands?

liggitt replied:
while you're in here, delete the else case for the GO_VERSION check above... code.google.com won't work any more, and that was intended for go1.3... the first branch should be used for go1.4+

liggitt commented on lib/vagrant-openshift/action/install_openshift_base_dependencies.rb:
```diff
@@ -110,6 +110,14 @@ def call(env)
+# TODO: Remove for go1.5
+# Check out a stable commit for go vet in order to version lock it to something we can work with
+pushd /data/src/golang.org/x/tools >/dev/null 2>&1
+git checkout c262de870b618eed648983aa994b03bc04641c72 
```
indent a couple spaces to make pushd/popd more readable

stevekuznetsov commented:
@liggitt Addressed comments.

liggitt commented on lib/vagrant-openshift/action/install_openshift_base_dependencies.rb:
```diff
@@ -104,11 +104,18 @@ def call(env)
-GOPATH=/data go get golang.org/x/tools/cmd/vet
+  GOPATH=/data go get golang.org/x/tools/cmd/vet
+
+  # TODO: Remove for go1.5
```
Move the todo up to the line with the "if 1.4*" check… that's what would need to be removed

stevekuznetsov replied:
This comment (poorly written) meant that `go vet` is moving into `go1.5` and none of this will be necessary on the upgrade. This won't apply to `go cover`.

liggitt commented:
LGTM

liggitt commented:
@danmcp merge?
